src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -5,20 +5,116 @@
  * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int buckets;
+    private Node[] table;
+
+    public MyHashMap() {
+        this.table = new Node[INITIAL_CAPACITY];
+        this.buckets = 0;
+        this.size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        addNode(key, value, table);;
+        resize();
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (size == 0) {
+            return null;
+        }
+        Node searchNode = table[indexFor(key, table.length)];
+        while (true) {
+            if (areKeysEqual(key, (K) searchNode.key)) {
+                return (V) searchNode.value;
+            }
+            if (searchNode.next == null) {
+                return null;
+            }
+            searchNode = searchNode.next;
+        }
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void addNode(K key, V value, Node[] table) {
+        int index = indexFor(key, table.length);
+        Node<K, V> node = new Node<>(getHash(key), key, value, null);
+        if (table[index] == null) {
+            table[index] = node;
+            size++;
+            buckets++;
+        } else {
+            Node searchNode = table[index];
+            while (true) {
+                if (areKeysEqual(key, (K) searchNode.key)) {
+                    searchNode.value = value;
+                    break;
+                } else if (searchNode.next == null) {
+                    searchNode.next = node;
+                    size++;
+                    break;
+                }
+                searchNode = searchNode.next;
+            }
+        }
+    }
+
+    private boolean areKeysEqual(K key1, K key2) {
```

Comment:
You don't need this method, your can use Objects.equals() instead(make sure you understand how it works)

Reply:
Wow! Null safe equals! :)